ocaml/xapi/xapi_cluster_helpers.ml

```diff
@@ -103,3 +103,60 @@ let with_cluster_operation ~__context ~(self : [`Cluster] API.Ref.t) ~doc ~op
           (Datamodel_common._cluster, Ref.string_of self)
       with _ -> ()
   )
+
+let maybe_generate_alert ~__context ~num_hosts ~missing_hosts ~new_hosts ~quorum
+    =
+  List.iter
+    (fun cluster_host ->
+      let host = Db.Cluster_host.get_host ~__context ~self:cluster_host in
+      let host_uuid = Db.Host.get_uuid ~__context ~self:host in
+      let host_name = Db.Host.get_name_label ~__context ~self:host in
+      let body =
+        Printf.sprintf
+          "Host %s has left the cluster, there are now %d host(s) in cluster \
+           and %d host(s) is required to form a quorum"
+          host_name num_hosts quorum
+      in
+      let name, priority = Api_messages.cluster_host_leaving in
+      Helpers.call_api_functions ~__context (fun rpc session_id ->
+          ignore
+          @@ Client.Client.Message.create ~rpc ~session_id ~name ~priority
+               ~cls:`Host ~obj_uuid:host_uuid ~body
+      )
+    )
```

Comment:
A lot of this code is repeated nearly identically twice below.

Reply:
ok, same `cluster_health` feature flag then?